README.md
# React Geosuggest [![Build Status](https://travis-ci.org/ubilabs/react-geosuggest.svg?branch=master)](https://travis-ci.org/ubilabs/react-geosuggest)

A [React](http://facebook.github.io/react/) autosuggest for the Google Maps Places API. You can also define your own suggests as defaults which continue to appear even during active input. Originally forked from [ubilabs-react-geosuggest](https://github.com/ubilabs/react-geosuggest).


## Demo

Live demo: [ubilabs.github.io/react-geosuggest](http://ubilabs.github.io/react-geosuggest/)


## Installation

As this component uses the Google Maps Places API to get suggests, you must include the Google Maps Places API in the `<head>` of your HTML:

```html
<!DOCTYPE html>
  <html>
  <head>
    …
    <script src="https://maps.googleapis.com/maps/api/js?libraries=places"></script>
  </head>
  <body>
    …
  </body>
</html>
```


The easiest way to use geosuggest is to install it from NPM and include it in your own React build process (using [Browserify](http://browserify.org), [Webpack](http://webpack.github.io/), etc).

```
npm install react-geosuggest-plus --save
```


## Usage

The Geosuggest works out of the box by just including it. However, you can customize the behaviour with the properties noted below.

```
var Geosuggest = require('react-geosuggest-plus');

<Geosuggest />
```

### Properties

#### autoActivateFirstSuggest
Type: `Boolean`
Default: `false`

Automatically activate the first suggestion as you type. If false, the exact term(s) in the input will be used when searching and may return a result not in the list of suggestions.

#### bounds
Type: [`LatLngBounds`](https://developers.google.com/maps/documentation/javascript/reference?csw=1#LatLngBounds)
Default: `null`

The bounds to use for biasing the suggests. If this is set, `location` and `radius` are ignored.

#### buttonMarkup
Type: `Function`
Default: `() => (<button>Enter</button>)`

The markup used for the submit button. This allows users to provide their own submit button. This is not rendered by default, unless the `showButton` property is set to `true`

#### className
Type: `String`
Default: `''`

Add an additional class to the geosuggest container.

#### country
Type: `String`
Default: `null`

Restricts predictions to the specified country (ISO 3166-1 Alpha-2 country code, case insensitive). E.g., us, br, au.

#### disabled
Type: `Boolean`
Default: `false`

Defines whether the input is disabled.

#### fixtures
Type: `Array`
Default: `[]`

An array with fixtures (defaults). Each fixture has to be an object with a `label` key in it. Optionally provide a `location`, but the Geosuggest will geocode the label if no location is provided.

You can also add a `className` key to a fixture. This class will be applied to the fixture item.

#### getRecentLabel
Type: `Function`
Default: `(suggest) => suggest.zipcode`

This function is used as an accessor for 'fixtures'. Since fixtures do not have a `label` property, this function is used to select the property that will be used instead. The default is the zipcode of the suggestion.

#### getSuggestLabel
Type: `Function`
Default: `(suggest) => suggest.description`

Used to generate a custom label for a suggest. Only parameter is a suggest (google.maps.places.AutocompletePrediction). [Check the Google Maps Reference](https://developers.google.com/maps/documentation/javascript/reference#GeocoderResult) for more information on it’s structure.

#### googleMaps
Type: `Object`
Default: `google.maps`

In case you want to provide your own Google Maps object, pass it in as googleMaps. The default is the global google maps object.

#### initialValue
Type: `String`
Default: `''`

An initial value for the input, when you want to prefill the suggest.

#### location
Type: `google.maps.LatLng`
Default: `null`

To get localized suggestions, define a location to bias the suggests.

#### noSuggestionsMarkup
Type: `Function`
Default: `() => {}`

A function that returns custom markup to be used when there are no available suggestions.

#### onBlur
Type: `Function`
Default: `(event) => {}`

Gets triggered when input field loses focus.

#### onChange
Type: `Function`
Default: `(event) => {}`

Gets triggered when input field changes the value

#### onClearClick
Type: `Function`
Default: `(event) => {}`

Gets triggered when the user 'clears' the input by pressing the clear button.

#### onClick
Type: `Function`
Default: `(event) => {}`

Gets triggered when the user clicks the input or the submit button.

#### onEmptySuggests
Type: `Function`
Default: `(event) => {}`

Gets triggered when the user hits the 'enter' button but no suggestions are selected.

#### onFocus
Type: `Function`
Default: `function() {}`

Gets triggered when the input field receives focus.

#### onSuggestSelect
Type: `Function`
Default: `function(suggest) {}`

Gets triggered when a suggest got selected. Only parameter is an object with data of the selected suggest. This data is available:

* `label` – Type `String` – The label name
* `placeId` – Type `String` – If it is a preset, equals the `label`. Else it is the Google Maps `placeID`
* `location` – Type `Object` – The location containing `lat` and `lng`
* `gmaps` – Type `Object` – *Optional!* The complete response when there was a Google Maps geocode necessary (e.g. no location provided for presets). [Check the Google Maps Reference](https://developers.google.com/maps/documentation/javascript/reference#GeocoderResult) for more information on it’s structure.

#### placeholder
Type: `String`
Default: `Search places`

The input field will get this placeholder text.

#### radius
Type: `Number`
Default: `0`

The radius defines the area around the location to use for biasing the suggests. It must be accompanied by a `location` parameter.

#### recentListItemMarkup
Type: `Component | Function`
Default: `<RecentListItem />`

The component used to display 'fixtures', or 'recents'. This defaults to the internal `<RecentListItem />` component.

Props
  - classes -- optional classes to pass to the component
  - onClick -- A function to call when the component is clicked (with the google suggest)
  - suggest -- The google suggestion

#### recentsLimit
Type: `Number`
Default: `5`

The max number of 'fixtures' to render.

#### showButton
Type: `Boolean`
Default: `false`

Whether or not to show the custom submit button.

#### skipSuggest
Type: `Function`
Default: `(suggest) => {}`

If the function returns true then the suggest will not be included in the displayed results. Only parameter is an object with data of the selected suggest. (See above)

#### types
Type: `Array`
Default: `null`

Restricts the suggestions against the passed in types. If none are specified, all types are used. Consult the Google Docs for [possible types](https://developers.google.com/places/supported_types).

### Exposed component functions

#### update(value)
It is possible to update the value of the input contained within the GeoSuggest component by calling the `update` function with a new desired `value` of the type String.

#### clear()
It is also possible to clear the value of the input contained within the GeoSuggest component by calling the `clear` function.

### Example

```
var React = require('react'),
  Geosuggest = require('./src/Geosuggest.jsx');

var App = React.createClass({
  /**
   * Render the example app
   */
  render: function() {
    var fixtures = [
      {label: 'Old Elbe Tunnel, Hamburg', location: {lat: 53.5459, lng: 9.966576}},
      {label: 'Reeperbahn, Hamburg', location: {lat: 53.5495629, lng: 9.9625838}},
      {label: 'Alster, Hamburg', location: {lat: 53.5610398, lng: 10.0259135}}
    ];

    return (
      <div>
        <Geosuggest
          placeholder="Start typing!"
          initialValue="Hamburg"
          fixtures={fixtures}
          onSuggestSelect={this.onSuggestSelect}
          location={new google.maps.LatLng(53.558572, 9.9278215)}
          radius="20" />
      </div>
    )
  },

  /**
   * When a suggest got selected
   * @param  {Object} suggest The suggest
   */
  onSuggestSelect: function(suggest) {
    console.log(suggest);
  }
});

React.render(<App />, document.getElementById('app'));
```

## Styling

This component uses [BEM](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) for namespacing the CSS classes. So styling should be easy and without conflicts. See the [geosuggest.css](https://github.com/ubilabs/react-geosuggest/blob/master/src/geosuggest.css) for an example styling.


## Contributing

### Development

To build the examples locally, run:

```
npm install
npm start
```

Then open [`localhost:8000`](http://localhost:8000) in a browser.

### Deployment

To release & deploy, run the following

```
npm run release:patch|minor|major
```


## License

The MIT License (MIT)
Copyright (c) 2015 Ubilabs <katzki@ubilabs.net>

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the “Software”), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.


CHANGELOG.md
## 1.2.0 (2017-02-21)


## 1.17.0 (2017-02-21)


## 1.17.0 (2017-02-21)


## 1.17.0 (2017-02-21)


## 1.17.0 (2017-02-21)


### 1.13.2 (2015-12-21)


### 1.13.1 (2015-12-21)


## 1.13.0 (2015-12-21)


## 1.13.0 (2015-10-29)

#### Features

* Remove defaults for search radius and location ([17f0bf0](https://github.com/ubilabs/react-geosuggest/commit/17f0bf053bde609c5f891ba4bb048d1cacc3f0b9))

## 1.12.0 (2015-10-22)


#### Bug Fixes

* doc comments, backward compatible changes ([96cca392](https://github.com/ubilabs/react-geosuggest/commit/96cca392f5b569247523cf784d34f1e1eb8b89d8))
* add copy css in build module ([bb988a1c](https://github.com/ubilabs/react-geosuggest/commit/bb988a1c5bf3079b170cd24c79bbd21093f49f93))
* googleMaps object moved out of props and set in componentDidMount ([955812df](https://github.com/ubilabs/react-geosuggest/commit/955812dfb702e8ea6318d89b2ef36d866c5c4354))


### 1.11.1 (2015-10-12)


#### Bug Fixes

* upgrade code to 0.14, too ([c5f64d47](https://github.com/ubilabs/react-geosuggest/commit/c5f64d47befef21c620bee1db41e86ffc1592194))


## 1.11.0 (2015-10-12)


#### Features

* bump peer dependency react to 0.14.0 ([dc39828e](https://github.com/ubilabs/react-geosuggest/commit/dc39828ea46552c60ce0a5ae33a52f6eee8c8f10))
* **input:** auto activate first suggest ([d1429b86](https://github.com/ubilabs/react-geosuggest/commit/d1429b8698c8928d69135fbe948f20f7e9246956))


## 1.10.0 (2015-09-29)


#### Bug Fixes

* prevent form submit on enter press ([dda86a12](https://github.com/ubilabs/react-geosuggest/commit/dda86a124a68ccf03220afb5f1796e99f183713e), closes [#49](https://github.com/ubilabs/react-geosuggest/issues/49))

#### Features

* add skipSuggest to not show certain suggestions  ([6da568d5](https://github.com/ubilabs/react-geosuggest/commit/6da568d5c0736fad7aacf21c864e8278544a544b)
* add getSuggestLabel to define a custom label  ([5103598](https://github.com/ubilabs/react-geosuggest/commit/51035989077a0eced308ac01f7e87a646893f767)
* **input:**  add disabled prop to disable ([7100d43e](https://github.com/ubilabs/react-geosuggest/commit/7100d43e2e3750e2506c78de32985974b915bb8f)

### 1.9.1 (2015-09-25)

#### Bug Fixes

* Fix issue if the initialValue changes ([b739b5c9](https://github.com/ubilabs/react-geosuggest/commit/b739b5c9b755f0efc05e28b01eb6b595b3d3cb9d))
* Build fixes so dist/ works with a global window.React  ([9b5e4369](https://github.com/ubilabs/react-geosuggest/commit/9b5e4369f7057e95ae5a36611cf5b7932dae50de))

## 1.9.0 (2015-09-09)

#### Features

* **input:**  add onChange callbacks  ([e6555ad](https://github.com/ubilabs/react-geosuggest/commit/e6555addbe3893b129488dc0623b7198036da35d)

## 1.8.0 (2015-09-01)

#### Features

* **input:**  add method to change the value of the user input  ([44d86f5](https://github.com/ubilabs/react-geosuggest/commit/44d86f5842765b72cb3db073feb016f750898e1f)


## 1.7.0 (2015-08-19)

#### Features

* **suggests:** add individual fixture classNames ([01b0e8a](https://github.com/ubilabs/react-geosuggest/commit/01b0e8a7a3e555729aeb56292a22bd8a412e4cf9)


## 1.6.0 (2015-08-12)


#### Features

* **input:** add bounds, country and types params ([00a84866](https://github.com/ubilabs/react-geosuggest/commit/00a84866d109ce4e323705558ffe319d56ecd5b1), closes [#20](https://github.com/ubilabs/react-geosuggest/issues/20))


## 1.5.0 (2015-08-12)


#### Features

* **input:**
  * add clear method to geosuggest  ([2d38a40](https://github.com/ubilabs/react-geosuggest/commit/2d38a4072b11c900d73b5cc26615a3cc69f286b4)
  * add onFocus and onBlur callbacks ([5051bc4](https://github.com/ubilabs/react-geosuggest/commit/5051bc424a4f508fa8dcff6683c690cc1ab9c2dd)

* **example:** add onFocus and onBlur demo ([88cf7f88](https://github.com/ubilabs/react-geosuggest/commit/88cf7f8873119f667aa514b8065e116d7a3741b2))


### 1.4.3 (2015-07-29)


#### Bug Fixes

* **example:** remove drop_console option from uglify ([e8e16112](https://github.com/ubilabs/react-geosuggest/commit/e8e16112d2d0f47f7ddc005dd485a25f7d55e4e7))


### 1.4.2 (2015-07-29)


#### Bug Fixes

* **build:** fix build for npm ([6475372](https://github.com/ubilabs/react-geosuggest/commit/6475372a468f19f075a20af012f8f85404172893))



### 1.4.1 (2015-07-21)


#### Bug Fixes

* **import:** add .npmignore ([add06073](https://github.com/ubilabs/react-geosuggest/commit/add06073990bb4c8934b8ad28c6c1bfb201f6945))


## 1.4.0 (2015-07-20)


#### Bug Fixes

* **input:** allow continuous editing ([62e83cce](https://github.com/ubilabs/react-geosuggest/commit/62e83cce9ac42e23916914691fb83b829d778d7e), closes [#12](https://github.com/ubilabs/react-geosuggest/issues/12))


#### Features

* **input:**
  * add new param className ([c1c990ec](https://github.com/ubilabs/react-geosuggest/commit/c1c990ec80e5210322d7d68b805d44a73196ca4e), closes [#8](https://github.com/ubilabs/react-geosuggest/issues/8))
  * add initial value param ([f97d8eba](https://github.com/ubilabs/react-geosuggest/commit/f97d8eba377ed789c1bbc21cfc4de94e85ef2760), closes [#13](https://github.com/ubilabs/react-geosuggest/issues/13))


test/setup.js
import jsdom from 'jsdom';

global.document = jsdom.jsdom('<!doctype html><html><body></body></html>');
global.window = document.defaultView;
global.navigator = window.navigator;

test/fixtures/recents.js
export default function createRecentsList() {
  return [
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
    {
      firstname: 'dustin',
      lastname: 'stiles',
      address1: '3 East 33rd Street',
      address2: '',
      city: 'New York',
      state_name: 'NY',
      place_name: '',
    },
  ];
}


src/GeosuggestItem.js
import React from 'react';

class GeosuggestItem extends React.Component {
  constructor(...args) {
    super(...args);

    this.handleClick = this.handleClick.bind(this);
    this.getSuggestClasses = this.getSuggestClasses.bind(this);
  }

  /**
   * The classes for the suggest item
   * @return {String} The classes
   */
  getSuggestClasses() {
    const className = this.props.suggest.className;
    let classes = 'geosuggest-item';

    classes += this.props.isActive ? ' geosuggest-item--active' : '';
    classes += className ? ' ' + className : ''; // eslint-disable-line

    return classes;
  }

  /**
   * When the element gets clicked
   * @param  {Event} event The click event
   */
  handleClick(event) {
    event.preventDefault();
    this.props.onSuggestSelect(this.props.suggest);
  }

  renderSaved() {
    const RecentListItem = this.props.recentListItemMarkup;

    return (
      <RecentListItem
        onClick={this.handleClick}
        suggest={this.props.suggest}
        classes={this.getSuggestClasses()}
      />
    );
  }

  renderGoogle() {
    return (
      <li className={this.getSuggestClasses()} onClick={this.handleClick}>
        {this.props.suggest.label}
      </li>
    );
  }
  /**
   * Render the view
   * @return {Function} The React element to render
   */
  render() {
    return this.props.suggest.label ? this.renderGoogle() : this.renderSaved();
  }

}

GeosuggestItem.defaultProps = {
  isActive: false,
  onSuggestSelect() {},
  suggest: {
    label: '',
  },
};

GeosuggestItem.propTypes = {
  isActive: React.PropTypes.bool,
  onSuggestSelect: React.PropTypes.func,
  recentListItemMarkup: React.PropTypes.func.isRequired,
  suggest: React.PropTypes.object,
};

export default GeosuggestItem;


example/react-geoex/app.js
/* global google */

var React = require('react'),
  ReactDOM = require('react-dom'),
  Geosuggest = require('../../src/Geosuggest'); // eslint-disable-line

var App = React.createClass({ // eslint-disable-line

  /**
   * When the input receives focus
   */
  onFocus: function() {
    console.log('onFocus sup'); // eslint-disable-line
  },

  /**
   * When the input loses focus
   */
  onBlur: function() {
    console.log('onBlur'); // eslint-disable-line
  },

  onChange: function(value) {
    console.log('input changes to :' + value); // eslint-disable-line
  },

  /**
   * When a suggest got selected
   * @param  {Object} suggest The suggest
   */
  onSuggestSelect: function(suggest) {
    console.log(suggest); // eslint-disable-line
  },

  /**
   * Render the example app
   * @return {Function} React render function
   */
  render: function() {
    var fixtures = [{
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '605 Park Ave',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': 'omgomg',
      'id': 23,
      'createdAt': '2014-12-03T12:55:17.000Z',
      'updatedAt': '2014-12-03T12:55:17.000Z'
    },
    {
      'firstname': 'Raheela',
      'lastname': 'Asifuddin',
      'place_name': null,
      'address1': '521 W 57th St',
      'address2': '3rd Floor',
      'city': 'New York',
      'state_name': 'NY',
      'zipcode': '10019',
      'phone': '(908) 507 - 3415',
      'special_instructions': 'It\'s her bday tomorrow! Thats all.',
      'id': 14934,
      'createdAt': '2015-06-25T15:22:28.000Z',
      'updatedAt': '2015-06-25T15:22:28.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '5 S Street NW Unit 1',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': 'sdfaasdf assadf ',
      'id': 12,
      'createdAt': '2014-12-03T11:16:16.000Z',
      'updatedAt': '2014-12-03T11:16:16.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '5 S Street NW Unit 1',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 16,
      'createdAt': '2014-12-03T11:31:31.000Z',
      'updatedAt': '2014-12-03T11:31:31.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '3300 16th Street NW #201',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': 'asdfasdfads asdf asdf asdf ',
      'id': 15,
      'createdAt': '2014-12-03T11:29:46.000Z',
      'updatedAt': '2014-12-03T11:29:46.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '605 Park Ave',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 22,
      'createdAt': '2014-12-03T12:43:19.000Z',
      'updatedAt': '2014-12-03T12:43:19.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '605 Park Ave',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 21,
      'createdAt': '2014-12-03T12:27:51.000Z',
      'updatedAt': '2014-12-03T12:27:51.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '5 S Street NW Unit 1',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': 'asdfa fasd asdf ',
      'id': 14,
      'createdAt': '2014-12-03T11:18:12.000Z',
      'updatedAt': '2014-12-03T11:18:12.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '5 S Street NW Unit 1',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 18,
      'createdAt': '2014-12-03T11:34:45.000Z',
      'updatedAt': '2014-12-03T11:34:45.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '5 S Street NW Unit 1',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 19,
      'createdAt': '2014-12-03T12:12:55.000Z',
      'updatedAt': '2014-12-03T12:12:55.000Z'
    },
    {
      'firstname': 'Chetan',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '122 E 65th St',
      'address2': '',
      'city': 'Manhattan',
      'state_name': 'NY',
      'zipcode': '10065',
      'phone': '2023025573',
      'special_instructions': '',
      'id': 20,
      'createdAt': '2014-12-03T12:17:12.000Z',
      'updatedAt': '2014-12-03T12:17:12.000Z'
    },
    {
      'firstname': 'Chai',
      'lastname': 'Shenoy',
      'place_name': null,
      'address1': '1111 20th Street Northwest',
      'address2': 'Peace Corps',
      'city': 'Washington',
      'state_name': 'DC',
      'zipcode': '20526',
      'phone': '(301) 520 - 7638',
      'special_instructions': '',
      'id': 10639,
      'createdAt': '2015-05-11T05:54:32.000Z',
      'updatedAt': '2015-05-11T05:54:32.000Z'
    },
    {
      'firstname': 'Test',
      'lastname': 'Order',
      'place_name': null,
      'address1': '987 Florida Ave NW',
      'address2': '',
      'city': 'Washington',
      'state_name': 'DC',
      'zipcode': '20001',
      'phone': '(999) 999 - 9999',
      'special_instructions': 'TEST; JUST DELETE',
      'id': 10338,
      'createdAt': '2015-05-08T17:25:09.000Z',
      'updatedAt': '2015-05-08T17:25:09.000Z'
    }
  ];

    return ( // eslint-disable-line
      <div>
        <Geosuggest
          fixtures={fixtures}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          onChange={this.onChange}
          onSuggestSelect={this.onSuggestSelect}
          getRecentLabel={(recent) => {
            return `${recent.address1}, ${recent.city}, ${recent.state_name} ${recent.zipcode}`;
          }}
          radius="20"
          location={new google.maps.LatLng(38.9047, -77.0164)}
        />
      </div>
    );
  }
});

ReactDOM.render(<App />, document.getElementById('app')); // eslint-disable-line